### DISCUSSION TITLE
this situation seems wrong

### DISCUSSION THREAD

**Discussion by wuxingping** [fabricjs/fabric.js#8684] [General]
the version is ^6.0.0-beta1. i  added fill and strokeWidth props into group class,but something is wrong.as follow:
![image](https://user-images.githubusercontent.com/41353266/218092821-9554a0dc-2e95-4cf1-98a5-c8e829dbb639.png)
`    var  shape = new Group([
new Rect({
        width: 100,
        height:100,
       fill:"red"
      }),new Rect({
      width: 200,
        height:200,
       fill:"pink"
      })
      ],
        {
          fill:"yellow",        
          stroke:"blue",
          strokeWidth:10,
          subTargetCheck:true,
          interactive:true,
          layout:"clip-path",  
          clipPath:rect,
        }
      )
`

i overwrided source code of  the version ^6.0.0-beta1.as follow:

`  drawObject(ctx: CanvasRenderingContext2D) {
         + (this.hasFill() || this.hasStroke()) && Rect.prototype._render.call(this, ctx);
          this._renderBackground(ctx);
          for (let i = 0; i < this._objects.length; i++) {
               this._objects[i].render(ctx);
          }
          this._drawClipPath(ctx, this.clipPath);
     }`

**Comment by ShaMan123** [comment]
I need a working demo
Seems `Rect._render` is mutating context. Try wrapping it with `ctx.save()...ctx.restore().`

**Reply by wuxingping** [reply]
i tried,but the problem still exists，the code as follow:
`drawObject(ctx: CanvasRenderingContext2D) {
          ctx.save();
          (this.hasFill() || this.hasStroke()) && Rect.prototype._render.call(this, ctx);         
          ctx.restore(); 
          this._renderBackground(ctx);
          for (let i = 0; i < this._objects.length; i++) {
               this._objects[i].render(ctx);
          }
          this._drawClipPath(ctx, this.clipPath);        
     }`

**Reply by wuxingping** [reply]
> ctx.save(); (this.hasFill() || this.hasStroke()) && Rect.prototype._render.call(this, ctx); ctx.restore()

**Reply by wuxingping** [reply]
it seems there is a conflict in the code `this._drawClipPath(ctx, this.clipPath)`. when i  delete the code, it work ok. but i cannot delete it.

on the hand , i  delete the code `this._drawClipPath(ctx, this.clipPath)`. the situation  as follow:
![1676281530563](https://user-images.githubusercontent.com/41353266/218424498-45d20997-355a-4abd-9216-251ac730411e.jpg)
on the other hand , when  i  deleted  the code `this._drawClipPath(ctx, this.clipPath)`. the situation is wrong when i drag the one inside group  as follow:
![image](https://user-images.githubusercontent.com/41353266/218424780-abc1b6b1-27b2-4884-b484-526d90501c24.png)

**Comment by asturur** [comment]
Try to reproduce the sistuation fully in a jsfillde, screenshot do not really help in this case.

**Reply by wuxingping** [reply]
> ok,i will try,thank you.
